### PR TITLE
Name validation, label generation widgets

### PR DESCRIPTION
**Pull request by tdtom167**
Fixing tests

### CODE REVIEW

File: src/databricksbundle/widgets/DatabricksWidgets.py [hunk 1]
```diff
@@ -2,14 +2,31 @@
...
+
 
 class DatabricksWidgets(Widgets):
     def __init__(self, dbutils: DBUtils):
         self.__dbutils = dbutils
         self.__multiselect_fields = []
+        self.__widget_index = 0
+        self.__name_validator = re.compile("^[a-z][a-z_0-9]+$")
+
+    def __generate_widget_label(self, name):
```
Comment (kutny) [inline src/databricksbundle/widgets/DatabricksWidgets.py]: Private methods after public ones pls.

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,4 @@ build
 dist/*
 .DS_Store
 !.gitkeep
+.vscode
```
Comment (kutny) [inline .gitignore:10]: please do not do this as part of this PR

**Review by kutny — CHANGES_REQUESTED** [review]

File: src/databricksbundle/widgets/DatabricksWidgetsTest.py [hunk 3]
```diff
@@ -124,6 +158,23 @@ def test_multiselect_single_value_invalid(self):
...
+    def test_multiselect_label_wrong_name(self):
+        with self.assertRaises(Exception) as error:
+            self.__widgets.add_multiselect("mymulti~", ["January", "February", "March"], ["January"])
+
+        self.assertEqual(
+            "The name you provided is incorrect, please provide name containing only alpha-numeric letters and _",
+            str(error.exception),
+        )
+
+    def test_multiselect_label_generation(self):
```
Comment (kutny) [inline src/databricksbundle/widgets/DatabricksWidgetsTest.py:170]: I do not see any multiselects in the test, only text fields

File: src/databricksbundle/widgets/DatabricksWidgetsTest.py [hunk 4]
```diff
@@ -86,25 +103,42 @@ def test_select_invalid_value(self):
...
-        self.assertEqual(["January", "February"], self.__widgets.get_value("mymulti"))
+        self.assertEqual(["January"], self.__widgets.get_value("mymulti"))
 
     def test_multiselect_defaults_overridden(self):
-        self.__widgets.add_multiselect("mymulti", ["January", "February", "March"], ["January", "February"])
+        self.__widgets.add_multiselect("mymulti", ["January", "February", "March"], ["January"])
         set_input(mymulti=["January", "March"])
         self.assertEqual(["January", "March"], self.__widgets.get_value("mymulti"))
 
     def test_multiselect_defaults_empty(self):
```
Comment (kutny) [inline src/databricksbundle/widgets/DatabricksWidgetsTest.py:134]: multiselect with no default values cannot be created?

**Review by kutny — CHANGES_REQUESTED** [review]

File: src/databricksbundle/widgets/DatabricksWidgetsTest.py (hunk 4 above)
Comment (tdtom167) [reply src/databricksbundle/widgets/DatabricksWidgetsTest.py:134]: I was not able to figure it out how to do it, therefore I believe not

File: src/databricksbundle/widgets/DatabricksWidgetsTest.py (hunk 3 above)
Comment (tdtom167) [reply src/databricksbundle/widgets/DatabricksWidgetsTest.py:170]: fixed

**Review by kutny — APPROVED** [review]

**Pull request merged**